src/util/TickUtils.ts

```diff
@@ -23,7 +23,19 @@ export function getTickBoundaries(viewBox: CartesianViewBox, sign: number, sizeK
   };
 }
 
-export function isVisible(sign: number, tickPosition: number, size: number, start: number, end: number): boolean {
+export function isVisible(
+  sign: number,
+  tickPosition: number,
+  getSize: () => number,
+  start: number,
+  end: number,
+): boolean {
+  if (sign * tickPosition < sign * start || sign * tickPosition > sign * end) {
+    return false;
+  }
```

Comment:
A very nice idea ❤️ 
Could you please add a small comment here for the motivation. It will help the next person working on this piece of the code.

Reply:
Sure! Let me know if the comment I added works 👍